File: 3430-count-days-without-meetings/count-days-without-meetings.java
class Solution {
    public int countDays(int days, int[][] meetings) {
        // Sort meetings by start time.
        Arrays.sort(meetings, (a, b) -> Integer.compare(a[0], b[0]));
        int ans = 0;
        int end = 0;
        
        // Process each meeting.
        for (int[] meeting : meetings) {
            int st = meeting[0];
            int fin = meeting[1];
            if (st > end) {
                ans += st - end - 1;
            }
            end = Math.max(end, fin);
        }
        
        // If there is a gap after the last meeting.
        if (days > end) {
            ans += days - end;
        }
        
        return ans;
    }
}

File: 3430-count-days-without-meetings/count-days-without-meetings.py
class Solution:
    def countDays(self, days: int, meetings: List[List[int]]) -> int:
        meetings.sort(key = lambda x : x[0])
        ans = 0
        end = 0

        for st, fin in meetings:
            if st > end:
                ans += st - end - 1
            end = max(end, fin)
        
        if days > end:
            ans += days - end
        
        return ans